core/services/ocr2/delegate.go

```diff
@@ -147,23 +147,25 @@ func (d Delegate) ServicesForSpec(jobSpec job.Job) (services []job.Service, err
 	jobSpec.PipelineSpec.JobName = jobSpec.Name.ValueOrZero()
 	jobSpec.PipelineSpec.JobID = jobSpec.ID
 
-	juelsPerFeeCoinPipelineSpec := pipeline.Spec{
-		ID:           jobSpec.ID,
-		DotDagSource: spec.JuelsPerFeeCoinPipeline,
-		CreatedAt:    time.Now(),
+	var pluginOracle plugins.OraclePlugin
+	switch spec.PluginType {
+	case job.Median:
+		pluginOracle, err = median.NewMedian(jobSpec, ocr2Provider, d.pipelineRunner, runResults, loggerWith, ocrLogger)
+	default:
+		return nil, fmt.Errorf("plugin type %s not supported", spec.PluginType)
 	}
-	numericalMedianFactory := median.NumericalMedianFactory{
-		ContractTransmitter: ocr2Provider.MedianContract(),
-		DataSource: ocrcommon.NewDataSourceV2(d.pipelineRunner,
-			jobSpec,
-			*jobSpec.PipelineSpec,
-			loggerWith,
-			runResults,
-		),
-		JuelsPerFeeCoinDataSource: ocrcommon.NewInMemoryDataSource(d.pipelineRunner, jobSpec, juelsPerFeeCoinPipelineSpec, loggerWith),
-		ReportCodec:               ocr2Provider.ReportCodec(),
-		Logger:                    ocrLogger,
+	if err != nil {
+		return nil, err
+	}
+	pluginFactory, err := pluginOracle.GetPluginFactory()
+	if err != nil {
+		return nil, err
+	}
+	pluginServices, err := pluginOracle.GetServices()
+	if err != nil {
+		return nil, err
 	}
+	services = append(services, pluginServices...)
```

Comment:
cleaner to just return all the services at the end instead of sprinkling appends e.g. 
```
return []job.Servce{ocr2Provider, oracle, ocrcommon.NewResultRunSaver(
		runResults,
		d.pipelineRunner,
		make(chan struct{}),
		loggerWith,
	), pluginServices...}
```